ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/handler/AbortedTxnCleaner.java

```diff
@@ -118,10 +120,16 @@ private void clean(CompactionInfo info, long minOpenWriteTxn, boolean metricsEna
       abortCleanUsingAcidDir(info, location, minOpenWriteTxn);
 
     } catch (InterruptedException e) {
+      LOG.error("Caught an interrupted exception when cleaning, unable to complete cleaning of {} due to {}", info,
+              e.getMessage());
+      info.errorMessage = e.getMessage();
+      handleCleanerAttemptFailure(info);
       throw e;
     } catch (Exception e) {
       LOG.error("Caught exception when cleaning, unable to complete cleaning of {} due to {}", info,
               e.getMessage());
+      info.errorMessage = e.getMessage();
+      handleCleanerAttemptFailure(info);
```

Comment:
minor, could we move `info.errorMessage = e.getMessage();` inside of handleCleanerAttemptFailure?

Reply:
This would have to pass the exception message to the `handleCleanerAttemptFailure()`. I dont see much merit in this.

Reply:
Done.